packages/zoe/src/contracts/automaticRefund.js

```diff
@@ -22,11 +22,12 @@ export const makeContract = harden(zcf => {
   const makeRefundInvite = () =>
     zcf.makeInvitation(refundOfferHook, 'getRefund');
 
-  return harden({
-    invite: makeRefundInvite(),
-    publicAPI: {
+  zcf.updatePublicAPI(
+    harden({
       getOffersCount: () => offersCount,
       makeInvite: makeRefundInvite,
-    },
-  });
+    }),
+  );
+
+  return harden(makeRefundInvite());
```

Comment:
This `harden` should be unnecessary. `makeRefundInvite()` should already return a hardened invite. In general Jessie "everything should be hardened" rules should harden objects where they're made, before they escape. Hence the `harden` on line 26 above around the object literal is required. The results of calls are not normally hardened by the caller, under the assumption that the callee is already returning a hardened value.

If a callee should not be relied upon to return a hardened value, then the value it returns should probably not be relied upon in other ways. It should probably be as suspect (or "alleged") as parameters normally are. Because this case is unusual, it should be called out explicitly in the code.

This is not one of those unusual cases.

Reply:
There are many such unnecessary `harden`s in this PR. Please fix all.

Reply:
Ok, thanks for pointing this out.